lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -681,15 +704,23 @@ def install_puppet_from_gem_on( host, opts )
           end
 
           if opts[:facter_version]
-            on host, "gem install facter -v'#{opts[:facter_version]}' --no-ri --no-rdoc"
+            on host, "gem install facter -v'#{opts[:facter_version]}' --no-ri --no-rdoc --no-format-executable"
           end
 
           if opts[:hiera_version]
-            on host, "gem install hiera -v'#{opts[:hiera_version]}' --no-ri --no-rdoc"
+            on host, "gem install hiera -v'#{opts[:hiera_version]}' --no-ri --no-rdoc --no-format-executable"
           end
 
           ver_cmd = opts[:version] ? "-v '#{opts[:version]}'" : ''
-          on host, "gem install puppet #{ver_cmd} --no-ri --no-rdoc"
+          on host, "gem install puppet #{ver_cmd} --no-ri --no-rdoc --no-format-executable"
+
+          # A gem install might not necessarily create these
+          ['confdir', 'logdir', 'codedir'].each do |key|
+            if host.puppet.has_key?(key)
+              dir = host.puppet[key]
+              on host, "[ -d #{dir} ] || mkdir -p #{dir}"
```

Comment:
This won't work on non-bash supported system.  Use host.mkdir_p(dir) so that the correct code is run on the correct platform.